### PR TITLE
feat(BUX-685): sql-injection safe filters

### PR DESCRIPTION
**Pull request by chris-4chain**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: engine/action_destination_test.go [hunk 1]
```diff
@@ -320,13 +322,12 @@ func (ts *EmbeddedDBTestSuite) TestClient_GetDestinations() {
 			require.NoError(t, err)
 			require.NotNil(t, destination)
 
-			trueValue := true
-
-			conditions := models.DestinationFilters{IncludeDeleted: &trueValue}
+			// deleted items should be present by default (empty conditions)
```
Comment (Nazarii-4chain) [inline engine/action_destination_test.go:325]: Thought we agreed to add `IncludeDeleted` flag if we want to include deleted records

File: models/filter/model_filter.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+// ToDbConditions converts filter fields to the datastore conditions using gorm naming strategy
+func (mf *ModelFilter) ToDbConditions() map[string]interface{} {
+	conditions := map[string]interface{}{}
+
+	applyConditionsIfNotNil(conditions, "created_at", mf.CreatedRange.ToDbConditions())
+	applyConditionsIfNotNil(conditions, "updated_at", mf.UpdatedRange.ToDbConditions())
+
+	if mf.IncludeDeleted == nil || !*mf.IncludeDeleted {
```
Comment (Nazarii-4chain) [inline models/filter/model_filter.go:17]: If we have 2 records. One active and second is deleted. With this condition we can get one of them or both ?

File: engine/action_destination_test.go (hunk 1 above)
Comment (chris-4chain) [reply engine/action_destination_test.go:325]: It's wrong comment - actually it works the opposite.

File: models/filter/model_filter.go (hunk 2 above)
Comment (chris-4chain) [reply models/filter/model_filter.go:17]: As the `IncludeDeleted` field says:
When we have `deleted-record` and `normal-record`
- when `IncludeDeleted` is set to true - we'll get `deleted-record` and `normal-record`
- when `IncludeDeleted` is nil or is set to false - we'll get only `normal-record`

**Review by Nazarii-4chain — APPROVED** [review]

**Pull request merged**